### PR TITLE
APE: Properly convert track/disk number pairs when writing

### LINKED ISSUE #159: APE track/disk numbers are not properly converted when writing
### Reproducer

I tried this code:

```rust
use lofty::{Accessor, Tag, TagExt, TagType};

fn main() {
    let mut tag = Tag::new(TagType::APE);
    tag.set_track(5);
    tag.set_track_total(15);
    tag.save_to_path("foo.mp3").unwrap();
}
```

### Summary

When using `Tag`, the track/disk numbers are not properly converted when saving to an APE tag. This is identical to #145, as it will be saved as two separate items: `........Track.5........Track.15`.

This issue was fixed for ID3v2 in #149, which stores number pairs in the same format. The fix should be similar, if not identical.

### Expected behavior

_No response_

### Assets

_No response_

### PR DESCRIPTION
**Pull request by Serial-ATA**
This also changes the default track/disk number to `DEFAULT_NUMBER_IN_PAIR`.

closes #159

### CODE REVIEW

File: src/id3/v2/util/pairs.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+		)),
+		(Some(number), Some(total)) => Some(format!("{number}{NUMBER_PAIR_SEPARATOR}{total}")),
+		(None, None) => None,
+	}
+}
+
+/// Attempts to convert a `TagItem` to a number, passing it to `setter`
+pub(crate) fn set_number<F: FnMut(u32)>(item: &TagItem, mut setter: F) {
+	let text = item.value().text();
+	let number = text.map(str::parse::<u32>);
```
Comment (uklotzde) [inline src/id3/v2/util/pairs.rs:39]: What happens if a field is empty or solely contains whitespace characters? These cases should probably be interpreted like a missing field.

The parsing should also not fail if the number is enclosed in leading/trailing whitespace characters, e.g. parsing  `" 12 "` should not fail.

**Review by uklotzde — CHANGES_REQUESTED** [review]
Parsing of numbers needs to be more robust and more liberal. The special cases should be tested.

File: src/id3/v2/util/pairs.rs (hunk 1 above)
Comment (uklotzde) [reply src/id3/v2/util/pairs.rs:39]: The text should be trimmed and then checked for emptiness before trying to parse the number.

File: Cargo.toml [hunk 2]
```diff
@@ -18,7 +18,7 @@ byteorder  = "1.4.3"
 # ID3 compressed frames
 flate2     = { version = "1.0.26", optional = true }
 # Proc macros
-lofty_attr = "0.8.0"
+lofty_attr = { path = "lofty_attr" }
```
[the remark this answers is not in the document]
Comment (Serial-ATA) [reply Cargo.toml:21]: Yes, it used `ape::tag::tagitems_into_ape`, which was changed.

File: src/id3/v2/util/pairs.rs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+			"{DEFAULT_NUMBER_IN_PAIR}{NUMBER_PAIR_SEPARATOR}{total}"
+		)),
+		(Some(number), Some(total)) => Some(format!("{number}{NUMBER_PAIR_SEPARATOR}{total}")),
+		(None, None) => None,
+	}
+}
+
+/// Attempts to convert a `TagItem` to a number, passing it to `setter`
+pub(crate) fn set_number<F: FnMut(u32)>(item: &TagItem, mut setter: F) {
+	let text = item.value().text().map(str::trim);
```
Comment (uklotzde) [inline src/id3/v2/util/pairs.rs:38]: Too many local variables with confusing naming. How about this:

```
let trimmed_text = item.value().text().unwrap_or_default().trim();
if trimmed_text.is_empty() {
    ...early return...
}
```

**Pull request merged**